lib/routemaster/redis_broker.rb

```diff
@@ -25,18 +27,42 @@ def cleanup
       _cleanup
     end
 
+    # Allow to inject pre-built Redis clients
+    #
+    # Before storing a new connection, ensure that any previously
+    # set client is properly closed.
+    #
+    def inject(clients={})
```

Comment:
I find this a little odd to inject here, I'm probably missing something but I would expect the `Config` class to allow overwriting the `drain_redis` and `cache_redis` when configuring instead of in the layer one down. Currently we have to check if we are managing the current connections in places, where if we allow it in the config it is clearer that we are not managing. 
What am I missing?